Firestore/core/test/firebase/firestore/util/executor_test.cc

```diff
@@ -105,6 +111,66 @@ TEST_P(ExecutorTest, DelayedOperationIsValidAfterTheOperationHasRun) {
   EXPECT_NO_THROW(delayed_operation.Cancel());
 }
 
+TEST_P(ExecutorTest, IsCurrentExecutor) {
+  EXPECT_FALSE(executor->IsCurrentExecutor());
+  EXPECT_NE(executor->Name(), executor->CurrentExecutorName());
+
+  executor->ExecuteBlocking([&] {
+    EXPECT_TRUE(executor->IsCurrentExecutor());
+    EXPECT_EQ(executor->Name(), executor->CurrentExecutorName());
+  });
+
+  executor->Execute([&] {
+    EXPECT_TRUE(executor->IsCurrentExecutor());
+    EXPECT_EQ(executor->Name(), executor->CurrentExecutorName());
+  });
+
+  Schedule(executor.get(), Executor::Milliseconds(1), [&] {
+    EXPECT_TRUE(executor->IsCurrentExecutor());
+    EXPECT_EQ(executor->Name(), executor->CurrentExecutorName());
+    signal_finished();
+  });
+
+  EXPECT_TRUE(WaitForTestToFinish());
+}
+
+TEST_P(ExecutorTest, ModifyingSchedule) {
```

Comment:
I'd suggest either a more specific test name or a descriptive comment about what the test is exercising. For example, it's not immediately clear if the interesting part is the time `chr::seconds(x)` or the tag (I think it's the latter).

If I'm understanding correctly maybe `TEST_P(ExecutorTest, PreservesDuplicates)`?

Reply:
This test was probably doing too much, split it into two -- one exercising `IsScheduled` and `PopFromSchedule`, the other specifically dealing with duplicate tags. Also added comments.